torch/distributed/checkpoint/_fsspec_filesystem.py

```diff
@@ -33,9 +35,28 @@ def create_stream(
         self, path: Union[str, os.PathLike], mode: str
     ) -> Generator[io.IOBase, None, None]:
         assert self.fs is not None
-        with self.fs.transaction:
-            with fsspec.open(str(path), mode) as stream:
+        # _open only supports binary mode
+        if "b" not in mode:
+            with self.create_stream(path, mode.replace("t", "") + "b") as stream:
+                yield io.TextIOWrapper(stream)
```

Comment:
Can you help me what is happening here? 
It seems like we don't have to handle the same discard/commit semantics in binary mode (?). Can you help me understand why? ( or a link to appropriate reading). thanks.

Reply:
> It seems like we don't have to handle the same discard/commit semantics in binary mode (?).

That's not quite right. The basic issue is that `fs._open` is only guaranteed to be implemented for binary mode -- so for text mode we just (recursively) get the binary stream (which has the discard + commit semantics) and then wraps it in an `io.TextIOWrapper`. The recursion ensures that the binary stream also has the same commit + discard semantics.

Reply:
This is right. It would have made sense for fsspec to wrap (heh) TextIOWrapper for "t" modes, to provide the same commit/discard methods.